### PR TITLE
WIP: Feat/frameless viewer

### PR DESCRIPTION
**Pull request by rjchow**
frameless viewer for mobile app integration

### CODE REVIEW

File: src/components/FramelessCertificateViewer.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+  const selectedTemplateName = get(certificate, "$template", "default");
+  const SelectedTemplate = templates[selectedTemplateName] || templates.default;
+
+  trace(`Templates Mapping: %o`, templates);
+  trace(`Selected template: ${selectedTemplateName}`);
+  trace(`Certificate content: %o`, certificate);
+
+  return (
+    <>
```
Comment (yehjxraymond) [inline src/components/FramelessCertificateViewer.js]: What is this for?

File: src/components/FramelessCertificateViewer.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      <div>
```
Comment (yehjxraymond) [inline src/components/FramelessCertificateViewer.js]: Do we need this too?

File: src/components/FramelessViewerPageContainer.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  }
+
+  handleCertificateChange(certificate) {
+    this.props.updateCertificate(certificate);
+  }
+
+  render() {
+    if (!this.props.document) {
+      return (
+        <>
```
Comment (yehjxraymond) [inline src/components/FramelessViewerPageContainer.js]: and this

File: src/components/FramelessViewerPageContainer.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    this.handleCertificateChange = this.handleCertificateChange.bind(this);
+    this.handleTextFieldChange = this.handleTextFieldChange.bind(this);
+  }
+
+  handleTextFieldChange(e) {
+    const fieldContents = JSON.parse(e.target.value);
+    trace(fieldContents);
+    const validated = validateSchema(fieldContents);
+    trace(`Certificate schema validation: ${validated}`);
+    const verified = verifySignature(fieldContents);
```
Comment (yehjxraymond) [inline src/components/FramelessViewerPageContainer.js]: both verifying sig and schema does not throw? The error wont get bubbled up right?

File: src/components/FramelessCertificateViewer.js (hunk 1 above)
Comment (rjchow) [reply src/components/FramelessCertificateViewer.js]: fixed

File: src/components/FramelessCertificateViewer.js (hunk 2 above)
Comment (rjchow) [reply src/components/FramelessCertificateViewer.js]: fixed

File: src/components/FramelessViewerPageContainer.js (hunk 3 above)
Comment (rjchow) [reply src/components/FramelessViewerPageContainer.js]: fixed

File: src/components/FramelessViewerPageContainer.js (hunk 4 above)
Comment (rjchow) [reply src/components/FramelessViewerPageContainer.js]: oops, verification is for informational purposes only but schema should throw. fixed

File: next.config.js [hunk 5]
```diff
@@ -20,7 +20,8 @@ module.exports = withBundleAnalyzer(
           "/registry": { page: "/registry" },
           "/privacy": { page: "/privacy" },
           "/viewer": { page: "/viewer" },
-          "/faq": { page: "/faq" }
+          "/faq": { page: "/faq" },
+          "/frameless_viewer": { page: "/frameless_viewer" }
```
Comment (yehjxraymond) [inline next.config.js]:
```suggestion
          "/frameless_viewer": { page: "/frameless-viewer" }
```

**Pull request merged**